pkg/feature/templates/servicemesh/kserve/kserve-predictor-authorizationpolicy.tmpl

```diff
@@ -0,0 +1,18 @@
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: kserve-predictor
+  namespace: {{ .ControlPlane.Namespace }}
+spec:
+  action: CUSTOM
+  provider:
+    name: opendatahub-odh-auth-provider
+  rules:
+  - to:
+    - operation:
+        notPaths:
+        - /healthz*
+        - /metrics
```

Comment:
@Xaenalt @VedantMahabaleshwarkar With the pod behind auth, we are leaving unprotected only these two paths/patterns.

I wonder if these are the only two that would need to remain open, or if there is anything else?

Reply:
I checked what we have on Queue-Proxy:

- 8022: `/wait-for-drain` (for a K8s hook)
- 9090: `/metrics`
- 8008: `/debug/pprof/` (if enabled)

Health checks (readiness-checks) are based on a header, so `/healthz*` is not going to match. But probably that is handled (rewritten by istio)?

```yaml
      readinessProbe:
        failureThreshold: 3
        httpGet:
          httpHeaders:
            - name: K-Network-Probe
              value: queue
          path: /
          port: 8012
          scheme: HTTP
        periodSeconds: 10
        successThreshold: 1
        timeoutSeconds: 1
```

Reply:
So, I adjusted this Policy to leave unprotected these three paths.

I'm a little bit unsure about `/metrics`. I'm not sure if that would be considered as sensitive data... Not sure if metrics and their labels would ever contain something.

@bartoszmajsak @aslakknutsen Not sure if we should create a secondary `DENY` policy that would reject access to `/metrics` if called from the outside.

BTW, I removed `/healthz*`. Looks like this one is for the istio sidecar, but seems to work without it being here.

> Health checks (readiness-checks) are based on a header, so /healthz* is not going to match. But probably that is handled (rewritten by istio)

@ReToCode Yes. And looks like istio lets these requests go in without them being here in this AuthorizationPolicy.

Reply:
`/metrics` is called from Knative as well. Do our control-plane components send auth contexts? Otherwise this should also be unprotected (same for prometheus scraping).

Reply:
> Do our control-plane components send auth contexts?

I think Nope. And yes, it's being left unprotected for now.

Reply:
For probes Istio rewrites them if you use for example: 
`sidecar.istio.io/rewriteAppHTTPProbers`, I think you use the latter correct?

Reply:
I saw it being rewritten by default. I guess we add `sidecar.istio.io/rewriteAppHTTPProbers` just in case it is disabled in the SMCP. But yes, we do use the rewrite in the odh-dashboard.